protocol/app/ante/pubkey.go

```diff
@@ -0,0 +1,110 @@
+package ante
```

Comment:
@teddyding I added a new decorator that looks identical to `SetPubKey` but only emits events 

reference from Osmosis ([link](https://github.com/osmosis-labs/osmosis/blob/2d938d0d31ea35c11286a46cc30f9a2e03997332/app/ante.go#L87))

Reply:
Nice, how did you figure this was necessary?

Reply:
Was just looking at the `SetPubKey` decorator and also using Osmosis as reference.